### ISSUE TITLE
support logical and operation for roles

### ISSUE THREAD

**Issue by matthiasbalke** [int128/gradle-ssh-plugin#198]
Currently I can't see a way to execute a ssh command on a node which has `roleA AND roleB` applied.

This could be used somehow like this:
```groovy
ssh.run {
  session(remotes.allRoles('roleA', 'roleB')) {
    // execute only on remotes with both roles
    execute "ls"
  }
}
```

**Comment by matthiasbalke** [comment]
I'll add a pull request for this as soon as I tested it locally.

**Comment by matthiasbalke** [comment]
How can I test the modified plugin version locally with my regular project configuration?

**Comment by int128** [comment]
Invoke `./gradlew publishMavenPublicationToMavenLocal` to install the modified version into the local maven repository and use it by:

```groovy
buildscript {
    repositories {
        mavenLocal()
        jcenter()
    }
    dependencies {
        classpath "org.hidetake:gradle-ssh-plugin:SNAPSHOT"
    }
}

apply plugin: 'org.hidetake.ssh'
```

**Comment by matthiasbalke** [comment]
Thanks for the fast support! Works as expected! Ready to merge!

**Comment by int128** [comment]
Thank you for the pull request! I merged your code into Groovy SSH and changed the plugin to use it from the plugin. `allRoles()` will be available in release 1.5.0.
